src/pylibsshext/logging.pyx

```diff
@@ -0,0 +1,94 @@
+#
+# This file is part of the pylibssh library
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, see file LICENSE.rst in this
+# repository.
+
+import contextlib
+import logging
+
+from pylibsshext.errors cimport LibsshSessionException
+from pylibsshext.includes cimport callbacks, libssh
+
+
+ANSIBLE_PYLIBSSH_TRACE = int(logging.DEBUG / 2)
+
+
+LOG_MAP = {
+    logging.NOTSET: libssh.SSH_LOG_TRACE,
+    ANSIBLE_PYLIBSSH_TRACE: libssh.SSH_LOG_TRACE,
+    logging.DEBUG: libssh.SSH_LOG_DEBUG,
+    logging.INFO: libssh.SSH_LOG_INFO,
+    logging.WARNING: libssh.SSH_LOG_WARN,
+    logging.ERROR: libssh.SSH_LOG_WARN,
+    logging.CRITICAL: libssh.SSH_LOG_NONE,
+}
```

Comment:
What if we use a sentinel object for "not set" and special-case it later? Plus expose an explicit "no log".
```suggestion
ANSIBLE_PYLIBSSH_NOLOG = float('inf')
ANSIBLE_PYLIBSSH_TRACE = int(logging.DEBUG / 2)

_NOT_SET_SENTINEL = object()


LOG_MAP = {
    logging.NOTSET: _NOT_SET_SENTINEL,
    ANSIBLE_PYLIBSSH_TRACE: libssh.SSH_LOG_TRACE,
    logging.DEBUG: libssh.SSH_LOG_DEBUG,
    logging.INFO: libssh.SSH_LOG_INFO,
    logging.WARNING: libssh.SSH_LOG_WARN,
    logging.ERROR: libssh.SSH_LOG_WARN,
    logging.CRITICAL: libssh.SSH_LOG_NONE,
    ANSIBLE_PYLIBSSH_NOLOG: libssh.SSH_LOG_NONE,
}
```

Reply:
Having seen your hint that `logging` might require an `int`, this could be
```suggestion
ANSIBLE_PYLIBSSH_NOLOG = logging.FATAL * 2
ANSIBLE_PYLIBSSH_TRACE = int(logging.DEBUG / 2)

_NOT_SET_SENTINEL = object()


LOG_MAP = {
    logging.NOTSET: _NOT_SET_SENTINEL,
    ANSIBLE_PYLIBSSH_TRACE: libssh.SSH_LOG_TRACE,
    logging.DEBUG: libssh.SSH_LOG_DEBUG,
    logging.INFO: libssh.SSH_LOG_INFO,
    logging.WARNING: libssh.SSH_LOG_WARN,
    logging.ERROR: libssh.SSH_LOG_WARN,
    logging.CRITICAL: libssh.SSH_LOG_NONE,
    ANSIBLE_PYLIBSSH_NOLOG: libssh.SSH_LOG_NONE,
}
```

Reply:
> Note that duplicate values will cause a smaller LOG_MAP_REV. Should we consider having a smaller dict for determinism and then augmenting it with additional mappings post-reversal?

If we remove the duplicates, we will fail to map in one of the ways. Note, that this maps X->Y and Y->X without any compicated logic like getting the first smaller/larger value. Yes, adding some logic to mapping can make the lists smaller by few items, but I think the complexity of the mapping/matching would not be worth the smaller map (unless there is some simple way to do that in python that I am not aware of).

Reply:
Currently, `libssh.SSH_LOG_WARN` maps to one of `logging.WARNING` or `logging.ERROR` in `LOG_MAP_REV`. It's probably predictable, but I can't say which one by just looking at code. If we only keep one in `LOG_MAP`, then produce `LOG_MAP_REV` (like it is now) and only then extend `LOG_MAP` to add “aliases”, it should give us better control over what's in the main mapping vs. the extended one.

Reply:
Thank you for the suggestion. This makes sense.